templates/storyboards/swift3.stencil

```diff
@@ -24,89 +20,74 @@ protocol StoryboardSceneType {
 }
 
 extension StoryboardSceneType {
-  static func storyboard() -> UIStoryboard {
-    return UIStoryboard(name: self.storyboardName, bundle: Bundle(for: BundleToken.self))
-  }
-
-  static func initialViewController() -> UIViewController {
-    guard let vc = storyboard().instantiateInitialViewController() else {
-      fatalError("Failed to instantiate initialViewController for \(self.storyboardName)")
-    }
-    return vc
+  static func storyboard() -> {{prefix}}Storyboard {
+    return {{prefix}}Storyboard(name: self.storyboardName, bundle: Bundle(for: BundleToken.self))
   }
 }
 
 extension StoryboardSceneType where Self: RawRepresentable, Self.RawValue == String {
```

Comment:
Right, but `destination` is an `UIViewController` instance, which doesn't have a `sceneIdentifier` property. Unfortunately controllers, once instantiated, can't tell where they come from (they have a `storyboard` property, but no identifier property).

In a `prepareForSegue`, you already get the instantiated view controller, no identifier. In `shouldPerformSegue`, you only get the segue identifier, no destination info.

Segues use identifiers at multiple points, so they should remain `enum`eratable.